python/ray/dashboard/client/src/pages/dashboard/Dashboard.tsx

```diff
@@ -47,26 +55,46 @@ class Dashboard extends React.Component<
     typeof mapDispatchToProps
 > {
   timeoutId = 0;
+  tabs = [
+    { label: "Machine view", component: NodeInfo },
+    { label: "Logical view", component: LogicalView },
+    { label: "Ray config", component: RayConfig },
+    { label: "Memory", component: MemoryInfo },
+    { label: "Tune", component: Tune },
+  ];
 
-  refreshNodeAndRayletInfo = async () => {
+  refershInfo = async () => {
+    let shouldObtainMemoryTable = this.props.shouldObtainMemoryTable;
     try {
-      const [nodeInfo, rayletInfo, tuneAvailability] = await Promise.all([
+      if (!shouldObtainMemoryTable) {
```

Comment:
If you follow my other suggestion of use effect hook in Memory table you can remove this.